typings/markup.d.ts

```diff
@@ -0,0 +1,172 @@
+import * as tt from './telegram-types.d'
+
+interface Button {
+  text: string
+  hide: boolean
+}
+
+interface ContactRequestButton {
+  text: string
+  hide: boolean
+  request_contact: boolean
+}
+
+interface LocationRequestButton {
+  text: string
+  hide: boolean
+  request_location: boolean
+}
+
+interface UrlButton {
+  url: string
+  text: string
+  hide?: boolean
+}
+
+interface CallbackButton {
+  text: string
+  hide: boolean
+  callback_data: string
+}
+
+interface SwitchToChatButton {
+  text: string
+  hide: boolean
+  switch_inline_query: string
+}
+
+interface SwitchToCurrentChatButton {
+  text: string
+  hide: boolean
+  switch_inline_query_current_chat: string
+}
+
+interface GameButton {
+  text: string
+  hide: boolean
+  callback_game: tt.CallbackGame
+}
+
+interface PayButton {
+  pay: boolean
+  text: string
+  hide: boolean
+}
+
+interface Buttons {
+  url?: string
+  pay?: boolean
+  text: string
+  callback_data?: string
+  callback_game?: string
+  switch_inline_query?: string
+  switch_inline_query_current_chat?: string
+}
```

Comment:
```suggestion
export type Buttons = Button | ContactRequestButton | LocationRequestButton | UrlButton | CallbackButton | SwitchToChatButton | SwitchToCurrentChatButton | GameButton | PayButton
```

This should be a simple 'any of the buttons' type

Reply:
I'll do it. Just tell me, what is the `hide` field in `interface Button`?

Reply:
When hide is true the buttons are not placed in the keyboard.
https://github.com/telegraf/telegraf/blob/252513a1e80aa444c88e2a33e056cf96023c61d2/markup.js#L184-L186

```ts
Markup.inlineKeyboard([
  [Markup.callbackButton('hey', 'x', Math.random() > 0.5)]
])
```
is the equivalent of
```ts
const buttons: Buttons = []
if (Math.random() > 0.5) {
  buttons.push([Markup.callbackButton('hey', 'x')])
}
Markup.inlineKeyboard(buttons)
```